### PR TITLE
Bug 1649822 - Rework Push Health Test Failure Layout

### PR DESCRIPTION
**Pull request by camd**
This adds the tabs for "Linting", "Builds", and "Tests" as well as a good portion of the other layout elements of the "Tests" section.  

A few things:

* This does **not** yet split the tests by "Crashes"  and "Test Failures".  I'll do that in a follow-up PR (it was making this large PR even larger).
* I also noticed that the caret next to the test path does not switch between pointed right and pointed down when you expand it.  I'll also fix that in a follow-up.
* The buttons for "group by" and "order by" I tried an outline, rather than a light grey.  To be honest, this was easier.  There's no default Bootstrap button color for light grey.  There is `color` of `light`, but it looks too light to me against a white background.  I could create a `lightgrey` color, but I'd have to handle all the hover color changes, etc.  The outline seemed like a good alternative.  But if you prefer, I'll create a light grey button CSS class.

Screenshot 2020-07-07 11 57 22

### CODE REVIEW

File: ui/push-health/details/DetailsPanel.jsx [hunk 1]
```diff
@@ -209,7 +209,7 @@ class DetailsPanel extends React.Component {
                     repoName={currentRepo.name}
                   />
                 </TabPanel>
-                <TabPanel>
+                <TabPanel className="overflow-auto h-100">
```
Comment (camd) [inline ui/push-health/details/DetailsPanel.jsx:212]: If the list of artifacts is longer than the height of the details panel, it was overflowing.  This is orthogonal to the rest of this PR, but figured I'd fix it while I was here.

**Comment by sarah-clements** [comment]
> * The buttons for "group by" and "order by" I tried an outline, rather than a light grey.  To be honest, this was easier.  There's no default Bootstrap button color for light grey.  There is `color` of `light`, but it looks too light to me against a white background.  I could create a `lightgrey` color, but I'd have to handle all the hover color changes, etc.  The outline seemed like a good alternative.  But if you prefer, I'll create a light grey button CSS class.

I think they look OK with the outline. One observation about that group of buttons: I think the idea was that they'd only show when expanded (so not showing them by default for known issues). I think it'd look a cleaner to do it that way.

**Comment by sarah-clements** [comment]
> > * The buttons for "group by" and "order by" I tried an outline, rather than a light grey.  To be honest, this was easier.  There's no default Bootstrap button color for light grey.  There is `color` of `light`, but it looks too light to me against a white background.  I could create a `lightgrey` color, but I'd have to handle all the hover color changes, etc.  The outline seemed like a good alternative.  But if you prefer, I'll create a light grey button CSS class.
> 
> I think they look OK with the outline. One observation about that group of buttons: I think the idea was that they'd only show when expanded (so not showing them by default for known issues). I think it'd look a cleaner to do it that way. Maybe that means it'd need to sit a little below the header instead of inline.

**Comment by camd** [comment]
Sounds good.  OK, those buttons are hidden when either of those groups are collapsed.

**Comment by sarah-clements** [comment]
@camd Something I just noticed in push health is that when you click an error line in the failures summary tab, it opens up a new log viewer page like the jobs view. But the way the mockup was designed is that clicking on that link switches to the logviewer tab. So I think it might be confusing to have both that tab and then have that link go to the logviewer in a new tab.

**Comment by sarah-clements** [comment]
Also this looks a little odd:
Screen Shot 2020-07-07 at 5 45 57 PM

Can you remove the spacing between the test suite and the tier/task in the next pr?

**Comment by camd** [comment]
Sure, I can fix that.

**Comment by camd** [comment]
> @camd Something I just noticed in push health is that when you click an error line in the failures summary tab, it opens up a new log viewer page like the jobs view. But the way the mockup was designed is that clicking on that link switches to the logviewer tab. So I think it might be confusing to have both that tab and then have that link go to the logviewer in a new tab.

Oh ok.  I hadn't noticed that behavior, but we can do that.  I'm going to open another bug for that since that interaction will take some work.  It'll be more than just a hyperlink since it will need to switch the tab, scan to the right line, and all that.

**Comment by sarah-clements** [comment]
> > @camd Something I just noticed in push health is that when you click an error line in the failures summary tab, it opens up a new log viewer page like the jobs view. But the way the mockup was designed is that clicking on that link switches to the logviewer tab. So I think it might be confusing to have both that tab and then have that link go to the logviewer in a new tab.
> 
> Oh ok. I hadn't noticed that behavior, but we can do that. I'm going to open another bug for that since that interaction will take some work. It'll be more than just a hyperlink since it will need to switch the tab, scan to the right line, and all that.

Sounds good - thanks :)

**Comment by camd** [comment]
Here's that bug, fwiw: https://bugzilla.mozilla.org/show_bug.cgi?id=1651408

The rest of these comments will be addressed in my next PR where I split `tests` and `crashes`.
